File: FoodDeliveryWebsite/FoodDeliveryWebsite.Models/Dtos/ProductDto.cs
﻿using FoodDeliveryWebsite.Models.Enums;

namespace FoodDeliveryWebsite.Models.Dtos
{
    public class ProductDto
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public decimal Price { get; set; }

        public ProductType Type { get; set; }

        public ProductStatus Status { get; set; }

        public int Grams { get; set; }

        // public byte[] Image { get; set; }
    }
}

File: angular-food-delivery-website/src/app/product-dto.ts
export class ProductDto {
    id: number
    name: string
    description: string
    price: number
    grams: number
    type: ProductType
    status: ProductStatus
    image: File
  }

  export enum ProductType {
    Salad = 1,
    Starter = 2,
    Main = 3,
    Seafood = 4,
    Bread = 5,
    Dessert = 6,
    Children = 7,
  }

  export enum ProductStatus {
    Available = 1,
    Unavailable = 2
  }
  
  export const ProductTypeEnumLocalization = {
    [ProductType.Salad]: 'Салата',
    [ProductType.Starter]: 'Предястие',
    [ProductType.Main]: 'Основно',
    [ProductType.Seafood]: 'Морски дарове',
    [ProductType.Bread]: 'Хляб',
    [ProductType.Dessert]: 'Десерт',
    [ProductType.Children]: 'Детско'
  };